examples/transform_and_produce/README.md

```diff
@@ -14,7 +16,26 @@ The process is fairly simple:
 - The credentials are then submitted to the `ProduceAndCommitStep` which simply produces the given message to `hash-topic`
 - The `ProduceAndCommitStep` is also responsible for commiting the original offset back for the consumer
 
-### Usage
+## Consume -> Batch -> Transform -> Unbatch -> Produce
+
+### What this does
+
+---
+
+This is a slightly more complex consumer that includes batching.
+
+The process is fairly simple:
+
+- The consumer consumes messages from the `raw-topic`
+- This message is in the form `{"username": "<username>", "password": "<password>"}`
+- Messages are batched
+- The batch is submitted to the `TransformStrategy` which would run some indexing logic on the whole
+  batch.
+- The results are exploded into messages again.
+- The indexed messages are then submitted to the `ProduceAndCommitStep` which simply produces the given message to `hash-topic`
```

Comment:
Does indexing/indexed have a standard definition in stream processing? This might be as clear as we need but a quick google search didn't immediately explain to me what "indexing logic" means

Reply:
Not really, that is just an example. We could replace indexing, with any other process we would do on batches, like storing fields, encode them into something, run a particle accelerator with that data, etc.